test/unit/DCAStrategies/dca-strategies-positions-handler.spec.ts

```diff
@@ -445,4 +456,211 @@ contract('DCAStrategiesPositionsHandler', () => {
       });
     });
   });
+
+  describe('syncPositionToNewVersion', () => {
+    let tx: TransactionResponse;
+    let totalAmount = ethers.utils.parseEther('50');
+    let delta = ethers.utils.parseEther('15');
+    let amountOfSwaps = BigNumber.from(5);
+    let newAmountOfSwaps = BigNumber.from(7);
+    let newPositions: BigNumber[];
+    let oldShares: IDCAStrategies.ShareOfTokenStruct[];
+    let newShares: IDCAStrategies.ShareOfTokenStruct[];
+
+    when('caller does not have permissions', () => {
+      given(async () => {
+        await DCAStrategiesPositionsHandlerMock.setPermissions(false);
+      });
+      then('tx reverts with message', async () => {
+        await expect(
+          DCAStrategiesPositionsHandlerMock.syncPositionToNewVersion(1, 3, user.address, user.address, amountOfSwaps, newAmountOfSwaps)
+        ).to.be.revertedWith('NoPermissions()');
+      });
+    });
+    when('syncPositionToNewVersion is called', () => {
+      given(async () => {
+        const positions = [1, 2, 3];
+        const OLD_SHARE_A = { token: tokenA.address, share: BigNumber.from(33e2) }; // reduce
+        const OLD_SHARE_B = { token: tokenB.address, share: BigNumber.from(33e2) }; // increase
+        const OLD_SHARE_C = { token: tokenC.address, share: BigNumber.from(34e2) }; // terminate
+        oldShares = sortTokens([OLD_SHARE_A, OLD_SHARE_B, OLD_SHARE_C]);
```

Comment:
Could we add a token whose shares doesn't change? It would be interesting to understand that it works correctly

Reply:
good catch, I wasn't accounting for a edge-case where nothing changes